### PR TITLE
we are pleased to announce the major update

### PR DESCRIPTION
**Pull request by rbesharat**
The GUI is redesigned using Pyqt5 in order to be able to use more advanced widgets.
There is the possibility of selecting all source platforms and entities at once.
There is a unified interface in which the source platform could be selected, rather than, a single version of the app per source platform.
The app is threaded so that it works faster and smoother.

### CODE REVIEW

File: procs/sqlite3/generate_selected_entities.py [hunk 1]
```diff
@@ -25,11 +25,12 @@
 }
 
 def handle_task(task, data_structure, task_proc_mapping):
-    try:
+    task_proc_mapping[task](data_structure)
+    """try:
```
Comment (tkiehn) [inline procs/sqlite3/generate_selected_entities.py:29]: is there a reason this block is commented out? @bvenedik-scalefree

File: backend/templates/ma_sat_v1.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{ config(schema='@@Schema', materialized='view') }} 
+
+{%- set yaml_metadata -%}
+sat_v0: '@@SatName'
+hashkey: '@@Hashkey'
+hashdiff: '@@Hashdiff' 
+ma_attribute: 
+@@MaAttribute
+ledts_alias: 'valid_to'
```
Comment (tkiehn) [inline backend/templates/ma_sat_v1.txt:9]: The default `ledts_alias` in datavault4dbt is `ledts`, is there a reason to deviate from that?

File: backend/templates/control_snap_v0.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{{ config(materialized='incremental') }}
+
+{%- set yaml_metadata -%}
+start_date: '2015-01-01'
+daily_snapshot_time: '07:30:00'
+{%- endset -%}    
+
+{%- set metadata_dict = fromyaml(yaml_metadata) -%}
+
+{%- set start_date = metadata_dict['start_date'] -%}
```
Comment (tkiehn) [inline backend/templates/control_snap_v0.txt:10]: Should switch to metadata_dict.get()

File: backend/templates/control_snap_v1.txt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    monthly:
+        duration: 5
+        unit: 'YEAR'
+    yearly:
+        forever: true
+{%- endset -%}    
+
+{%- set metadata_dict = fromyaml(yaml_metadata) -%}
+
+{%- set control_snap_v0 = metadata_dict['control_snap_v0'] -%}
```
Comment (tkiehn) [inline backend/templates/control_snap_v1.txt:21]: Should switch to metadata_dict.get()

File: backend/templates/nh_link.txt [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+source_models: @@SourceModels
+link_hashkey: @@LinkHashkey 
+foreign_hashkeys: @@ForeignHashkeys
+@@Payload
+{%- endset -%}
+
+
+{% set metadata_dict = fromyaml(yaml_metadata) %}
+
+{%- set link_hashkey = metadata_dict['link_hashkey'] -%}
```
Comment (tkiehn) [inline backend/templates/nh_link.txt:14]: Should switch to metadata_dict.get()

File: backend/templates/ref_hub.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{{ config(materialized='incremental',
+          schema='Core') }}
+
+{%- set yaml_metadata -%}
+source_models: @@SourceModel
+{%- endset -%}      
+
+{% set metadata_dict = fromyaml(yaml_metadata) %}
+
+{{ datavault4dbt.ref_hub(source_models=metadata_dict['source_models'],
```
Comment (tkiehn) [inline backend/templates/ref_hub.txt:10]: Should switch to metadata_dict.get()

File: backend/templates/ref_sat_v0.txt [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+{%- set yaml_metadata -%}
+source_model: @@SourceModel
+parent_ref_keys: @@RefKeys
+src_hashdiff: @@HashDiff
+src_payload: @@Payload
+{%- endset -%}      
+
+{% set metadata_dict = fromyaml(yaml_metadata) %}
+
+{{ datavault4dbt.ref_sat_v0(source_model=metadata_dict['source_model'],
```
Comment (tkiehn) [inline backend/templates/ref_sat_v0.txt:13]: Should switch to metadata_dict.get()

File: backend/templates/ref_sat_v1.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+{%- set yaml_metadata -%}
+ref_sat_v0: @@RefSat
+ref_keys: @@RefKeys
+hashdiff: @@HashDiff
+add_is_current_flag: true
+{%- endset -%}      
+
+{% set metadata_dict = fromyaml(yaml_metadata) %}
+
+{{ datavault4dbt.ref_sat_v1(ref_sat_v0=metadata_dict['ref_sat_v0'],
```
Comment (tkiehn) [inline backend/templates/ref_sat_v1.txt:13]: Should switch to metadata_dict.get()

File: backend/templates/ref_table.txt [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+{%- set yaml_metadata -%}
+ref_hub: '@@RefHub'
+ref_satellites: @@RefSat
+historized: '@@Historized'
+
+{%- endset -%}
+
+{% set metadata_dict = fromyaml(yaml_metadata) %}
+
+{{ datavault4dbt.ref_table(ref_hub=metadata_dict['ref_hub'],
```
Comment (tkiehn) [inline backend/templates/ref_table.txt:12]: Should switch to metadata_dict.get()

File: procs/sqlite3/generate_selected_entities.py (hunk 1 above)
Comment (bvenedik-scalefree) [reply procs/sqlite3/generate_selected_entities.py:29]: We moved the equivalent process of this try except block to a different location. Was commented out during development. I suppose the commented block can be removed.

File: backend/templates/ma_sat_v1.txt (hunk 2 above)
Comment (bvenedik-scalefree) [reply backend/templates/ma_sat_v1.txt:9]: Not sure about this one. Do you have any idea about this @rbesharat ?

**Pull request merged**